### PR TITLE
Implementation of bionic base stack

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

#1 
See comment below for diff between this stack and previous bionic base stack releases.
Ping me when this is merged in, because we will need to disable workflows on https://github.com/paketo-buildpacks/base-stack-release to avoid duplicate release publishing

I have already added a new tag to the repo with 1.2.4 so that the first release is versioned correctly

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by sophiewigmore** [comment]
<details>
<summary> build image diff output </summary>

```
-----File-----
container-diff diff  paketobuildpacks/build:1.2.3-base-cnb daemon://bionic-base-build:latest  --type file

These entries have been added to paketobuildpacks/build:1.2.3-base-cnb: None

These entries have been deleted from paketobuildpacks/build:1.2.3-base-cnb:
FILE                          SIZE
/etc/group-                   446B
/etc/gshadow-                 374B
/etc/passwd-                  926B
/etc/shadow-                  501B
/etc/subgid-                  0
/etc/subuid-                  0
/home/cnb/.bash_logout        220B
/home/cnb/.bashrc             3.7K
/home/cnb/.profile            807B

These entries have been changed between paketobuildpacks/build:1.2.3-base-cnb and bionic-base-build:latest:
FILE                                 SIZE1         SIZE2
/var/log/lastlog                     285.4K        28.8K
/var/log/dpkg.log                    246.6K        246.6K
/var/log/tallylog                    62.6K         6.3K
/var/log/faillog                     31.3K         3.2K
/var/log/apt/term.log                25.9K         25.9K
/var/log/apt/history.log             12.6K         12.6K
/var/cache/ldconfig/aux-cache        8.2K          8.2K
/var/log/alternatives.log            6.3K          6.3K
/etc/shadow                          526B          501B
/etc/gshadow                         382B          374B
/etc/apt/sources.list                225B          241B
/etc/default/locale                  92B           92B
/etc/subuid                          17B           0
/etc/subgid                          17B           0
```

</details>


<details>
<summary> run image diff output</summary>

```
 container-diff diff  paketobuildpacks/run:1.2.3-base-cnb daemon://bionic-base-run:latest  --type file | tee pbcopy

-----File-----

These entries have been added to paketobuildpacks/run:1.2.3-base-cnb: None

These entries have been deleted from paketobuildpacks/run:1.2.3-base-cnb:
FILE                          SIZE
/etc/group-                   446B
/etc/gshadow-                 374B
/etc/passwd-                  926B
/etc/shadow-                  501B
/etc/subgid-                  0
/etc/subuid-                  0
/home/cnb/.bash_logout        220B
/home/cnb/.bashrc             3.7K
/home/cnb/.profile            807B
/usr/lib/os-release           386B

These entries have been changed between paketobuildpacks/run:1.2.3-base-cnb and bionic-base-run:latest:
FILE                                 SIZE1         SIZE2
/var/log/lastlog                     285.4K        28.8K
/var/log/dpkg.log                    203.4K        203.4K
/var/log/tallylog                    62.6K         6.3K
/var/log/faillog                     31.3K         3.2K
/var/log/apt/history.log             9K            9K
/var/log/apt/term.log                6.9K          6.9K
/var/cache/ldconfig/aux-cache        5.3K          5.3K
/etc/shadow                          526B          501B
/etc/gshadow                         382B          374B
/etc/apt/sources.list                225B          241B
/etc/default/locale                  92B           92B
/etc/os-release                      21B           509B
/etc/subgid                          17B           0
/etc/subuid                          17B           0
```

</details>


The diffs in both the build and run images are of the same types [as those outlined by Frankie in the Bionic Tiny stack build image](https://github.com/paketo-buildpacks/bionic-tiny-stack/pull/1#issue-1283819805)

**Comment by sophiewigmore** [comment]
The Base Builder smoke tests using the build/run image generated from this code base succeeded, giving confidence that any changes to the stack are not breaking.

**Comment by fg-j** [comment]
@sophiewigmore Any guess on why `/usr/lib/os-release` disappeared from the build image?

File: metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+			image, err := index.Image(indexManifest.Manifests[0].Digest)
+			Expect(err).NotTo(HaveOccurred())
+
+			file, err := image.ConfigFile()
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(file.Config.Labels).To(SatisfyAll(
+				HaveKeyWithValue("io.buildpacks.stack.id", "io.buildpacks.stacks.bionic"),
+				HaveKeyWithValue("io.buildpacks.stack.description", "ubuntu:bionic + openssl + CA certs + compilers + shell utilities"),
```
Comment (fg-j) [inline metadata_test.go:74]: Could we take this opportunity to make the stack description more similar to the [updated bionic tiny stack description](https://github.com/paketo-buildpacks/bionic-tiny-stack/blob/65a4a1700d7bc0bac14d8539d117003041a76adb/stack/stack.toml#L8)?

File: metadata_test.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+			image, err := index.Image(indexManifest.Manifests[0].Digest)
+			Expect(err).NotTo(HaveOccurred())
+
+			file, err := image.ConfigFile()
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(file.Config.Labels).To(SatisfyAll(
+				HaveKeyWithValue("io.buildpacks.stack.id", "io.buildpacks.stacks.bionic"),
+				HaveKeyWithValue("io.buildpacks.stack.description", "ubuntu:bionic + openssl + CA certs"),
```
Comment (fg-j) [inline metadata_test.go:163]: likewise here

**Comment by sophiewigmore** [comment]
> @sophiewigmore Any guess on why /usr/lib/os-release disappeared from the build image?

Yeah, we're removing it on purpose in [the run Dockerfile](https://github.com/paketo-buildpacks/bionic-base-stack/blob/7fddcaefbde8a486a4e67bdeabd675130b381f89/stack/run.Dockerfile#L24-L29). This was straight up taken from the Jammy base stack. On the run images we currently publish, the `/usr/lib/os-release` and `/etc/os-release` files are identical. I don't know if it's a huge deal to be honest, I'm assuming it ended up in there because the tiny  stack does something like this, and Jammy base is largely copied from Jammy tiny. 

I can take those lines out if you want. I don't think the content in them is that important

**Comment by sophiewigmore** [comment]
> @sophiewigmore Any guess on why /usr/lib/os-release disappeared from the build image?

Do you mean the run image? If so, we're removing it in [the run Dockerfile](https://github.com/paketo-buildpacks/bionic-base-stack/blob/7fddcaefbde8a486a4e67bdeabd675130b381f89/stack/run.Dockerfile#L24-L29). This was straight up taken from the Jammy base stack, so that's why the lines are here specifically. On the run images we currently publish, the `/usr/lib/os-release` and `/etc/os-release` files are identical. I don't know if it's a huge deal to be honest, I'm assuming it ended up in there because the tiny stack does something like this, and Jammy base is largely copied from Jammy tiny.

I can take those lines out if you want, although I don't think the content in them is that important

File: buildpack_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+		syftBuildpack, err = buildpackStore.Get.
+			WithVersion("1.12.0").
+			Execute("github.com/paketo-buildpacks/syft")
+		Expect(err).NotTo(HaveOccurred())
+
+		executableJarBuildpack, err = buildpackStore.Get.
+			WithVersion("6.2.4").
+			Execute("github.com/paketo-buildpacks/executable-jar")
+		Expect(err).NotTo(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (sophiewigmore) [reply buildpack_integration_test.go:71]: copy pasta from jammy base stack, where this is also a problem. I will put in a PR to fix it.

File: metadata_test.go (hunk 1 above)
Comment (sophiewigmore) [reply metadata_test.go:74]: done

**Comment by fg-j** [comment]
^^ PS it looks like the "automatically delete branches" setting isn't set up on this repo. I forgot during initial repo setup!

**Pull request merged**